### PR TITLE
Config: simplify JSON validation and make it typecheck

### PR DESCRIPTION
**Pull request by jenshnielsen**
Use types-jsonschema and update the code to validate. 

In the process refactor the validation method to avoid nesting logic and make it more clear

### CODE REVIEW

File: qcodes/configuration/config.py [hunk 1]
```diff
@@ -175,20 +175,30 @@ def _update_config_from_file(
...
         Args:
-            json_config: json file to validate
+            json_config: json dictionary to validate
             schema: schema dictionary
             extra_schema_path: schema path that contains extra validators to be
                 added to schema dictionary
         """
+        if schema is None:
+            if self.current_schema is None:
+                raise RuntimeError("Cannot validate as " "current_schema is None")
```
Comment (astafan8) [inline qcodes/configuration/config.py:196]:
```suggestion
                raise RuntimeError("Cannot validate as current_schema is None")
```

**Review by astafan8 — APPROVED** [review]

File: qcodes/configuration/config.py (hunk 1 above)
Comment (jenshnielsen) [reply qcodes/configuration/config.py]: Black ;(

**Comment by astafan8** [comment]
bors merge

**Comment by jenshnielsen** [comment]
bors merge

**Pull request merged**